### ISSUE TITLE
Difference to arrays with respect to indexing with `[]` 

### ISSUE THREAD

**Issue by mschauer** [JuliaArrays/StaticArrays.jl#604]
I just noticed the discrepancy
```
julia> (@SVector [1])[]
ERROR: ArgumentError: invalid index: () of type Tuple{}
```

```
julia> ([1])[]
1
```

**Comment by mschauer** [comment]
This allowed in Base by https://docs.julialang.org/en/v1.3-dev/manual/arrays/#Omitted-and-extra-indices-1
 "Indices may be omitted if the trailing dimensions that are not indexed into are all length one." (via @mbauman )

**Comment by mschauer** [comment]
Related and partial(?) fix: #580 .

**Comment by c42f** [comment]
Fixed in #581

**Comment by mschauer** [comment]
Yes, fixed:
```

julia> m = @MArray (rand(1,1,1))
1×1×1 MArray{Tuple{1,1,1},Float64,3,1}:
[:, :, 1] =
 0.10877107273924591

julia> m[]
0.10877107273924591

```
